### ISSUE TITLE
The discriminator seems not to have been updated

### ISSUE THREAD

**Issue by Godev36** [Supersirui/TransStego#1]
It seems that there is no effective update of the discriminator parameters in the code you provided to make the discriminator more powerful. The specific code is:`                if not args.no_gan:
                    optimize_dis.zero_grad()
                    optimize_dis.step()`
I think the code should be modified to: `               if not args.no_gan:
                    optimize_dis.zero_grad()
                    Wasserstein_D_loss.backward()  # Wasserstein D loss is a negative number of Wasserstein D
                    optimize_dis.step()`
